ddl/reorg.go

```diff
@@ -154,22 +153,22 @@ func (d *ddl) isReorgRunnable(txn kv.Transaction) error {
 	return nil
 }
 
-func (d *ddl) delKeysWithPrefix(prefix string) error {
+func (d *ddl) delKeysWithPrefix(prefix kv.Key) error {
 	for {
-		keys := make([]string, 0, maxBatchSize)
+		keys := make([]kv.Key, 0, maxBatchSize)
 		err := kv.RunInNewTxn(d.store, true, func(txn kv.Transaction) error {
 			if err1 := d.isReorgRunnable(txn); err1 != nil {
 				return errors.Trace(err1)
 			}
 
-			iter, err := txn.Seek([]byte(prefix))
+			iter, err := txn.Seek(prefix)
 			if err != nil {
 				return errors.Trace(err)
 			}
 
 			defer iter.Close()
 			for i := 0; i < maxBatchSize; i++ {
-				if iter.Valid() && strings.HasPrefix(iter.Key(), prefix) {
+				if iter.Valid() && iter.Key().HasPrefix(prefix) {
 					keys = append(keys, iter.Key())
```

Comment:
is it safe to only append slice but not clone it ? 
next iterator may change the key buffer?